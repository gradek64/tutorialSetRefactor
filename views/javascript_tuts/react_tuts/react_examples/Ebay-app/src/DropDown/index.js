import React from 'react';
import PropTypes from 'prop-types';
import SimpleButton from '../Simplebutton/index';
import ToggleState from '../logic/ToggleState';
import ClickAwayListener from '../ClickAwayListener';

/*
  *@ToggleState is class componet that holds only logic and uses
  *@props.render method to exposed its renderMethod and toglle state
  *@SimpleButton is extending ToggleState (using its logic) and then whole thing (ToggleState and Simple button)
  *@is wrapped in ToggleSimpleButton functional component that is expoerted outside and takes props for SingleButton
*/
const ToggleSimpleButton = (props) => {
  /* eslint-disable no-lone-blocks */
  {
    /* toggleMethod is exposed from ToggleState class and allows to as well as toggle state
true or false , that could be used in other componenst as Simple button */
  }
  return (
    <ToggleState
      renderComp={(toggleMethod, toggle) => (
        <React.Fragment>
          {toggle ? <ClickAwayListener clickAway={toggleMethod} /> : null}
          <div
            className="dropdown"
            style={{ position: 'relative', display: 'flex' }}
          >
            <SimpleButton
              label={props.label}
              eventHandler={() => {
                toggleMethod(toggle);
              }}
            />
            {toggle ? (
              <div
                role="presentation"
                onClick={() => {
                  toggleMethod(toggle);
                }}
                // required by airbnb but div in this case can not be in focus
                onKeyPress={() => {}}
                style={{
                  position: 'absolute',
                  left: '0',
                  top: '0',
                }}
              >
                {props.children}
              </div>
            ) : null}
          </div>
        </React.Fragment>
      )}
    />
  );
};

ToggleSimpleButton.propTypes = {
  children: PropTypes.instanceOf(Object).isRequired,
  label: PropTypes.string.isRequired,
};

export default ToggleSimpleButton;
