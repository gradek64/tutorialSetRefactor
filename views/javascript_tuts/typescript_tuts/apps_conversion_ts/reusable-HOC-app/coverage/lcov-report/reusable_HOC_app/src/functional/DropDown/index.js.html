<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for reusable_HOC_app/src/functional/DropDown/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">reusable_HOC_app/src/functional/DropDown</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import SimpleButton from '../Simplebutton/index';
import ToggleState from '../../logic/ToggleState';
import ClickAwayListener from '../ClickAwayListener';
&nbsp;
/*
  *@ToggleState is class componet that holds only logic and uses
  *@props.render method to exposed its renderMethod and toglle state
  *@SimpleButton is extending ToggleState (using its logic) and then whole thing (ToggleState and Simple button)
  *@is wrapped in ToggleSimpleButton functional component that is expoerted outside and takes props for SingleButton
*/
const ToggleSimpleButton = (props) =&gt; {
  {
    /* toggleMethod is exposed from ToggleState class and allows to as well as toggle state
true or false , that could be used in other componenst as Simple button */
  }
  return (
    &lt;ToggleState
      render={(toggleMethod, toggle) =&gt; (
        &lt;React.Fragment&gt;
          {toggle ? &lt;ClickAwayListener clickAway={toggleMethod} /&gt; : null}
          &lt;div
            className="dropdown"
            style={{ position: 'relative', display: 'flex' }}
          &gt;
            &lt;SimpleButton
              label={props.label}
              action={() =&gt; {
                toggleMethod(toggle);
              }}
            /&gt;
            {toggle ? (
              &lt;div
                role="presentation"
                onClick={() =&gt; {
                  toggleMethod(toggle);
                }}
                // required by airbnb but div in this case can not be in focus
                onKeyPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; {}}
                style={{
                  position: 'absolute',
                  left: '0',
                  top: '0',
                }}
              &gt;
                {props.children}
              &lt;/div&gt;
            ) : null}
          &lt;/div&gt;
        &lt;/React.Fragment&gt;
      )}
    /&gt;
  );
};
&nbsp;
ToggleSimpleButton.propTypes = {
  children: PropTypes.instanceOf(Object).isRequired,
  label: PropTypes.string.isRequired,
};
&nbsp;
export default ToggleSimpleButton;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 18 2019 13:39:36 GMT+0000 (GMT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
