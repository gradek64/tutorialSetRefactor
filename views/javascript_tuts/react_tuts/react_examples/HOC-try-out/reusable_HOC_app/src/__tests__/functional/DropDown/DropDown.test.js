import React from 'react';
// full render needed
import { mount } from 'enzyme';

/*
  *@this next line is defined in a root/jest.configuration.js
  *@so no need to import library
*/
// import toJson from 'enzyme-to-json';

// component for snapshot test;
import DropDown from '../../../functional/DropDown';

function DropDownContentExample() {
  return (<div className="DropDownContent"><button>1</button><button>2</button></div>);
}

/*
  *@DropDown component needs children prop as required the way is provided is
  *@simply put JSX content between <DropDown>{props.children}</DropDown>
*/

let wrapper;
beforeAll(() => {
  wrapper = mount(<DropDown label="DropDown"><DropDownContentExample /></DropDown>);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('changes toggle after SimpleButton click', () => {
  const SimpleButton = wrapper.find('SimpleButton').find('button');
  // before
  expect(wrapper.find('DropDownContentExample').length).toBe(0);

  SimpleButton.simulate('click');
  // re-render component from Enzyme;
  wrapper.update();

  expect(wrapper.find('DropDownContentExample').length).toBe(1);
});

/*
  *@Test work in sequance that means changes from previous test will pass
  *@to next text , therefore our DropDown content is present for next text;
*/

test('DropDown goes away once content is clicked', () => {
  // quick assertion to prove comment above;
  expect(wrapper.find('DropDownContentExample').length).toBe(1);

  wrapper.find('DropDownContentExample').simulate('click');
  // update compoent;
  wrapper.update();
  // no more DropDownContentExample;
  expect(wrapper.find('DropDownContentExample').length).toBe(0);
});
