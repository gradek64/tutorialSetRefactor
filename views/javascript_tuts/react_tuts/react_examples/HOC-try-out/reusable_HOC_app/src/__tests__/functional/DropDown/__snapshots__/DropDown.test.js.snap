// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ToggleSimpleButton
  label="DropDown"
>
  <ToggleState
    render={[Function]}
  >
    <div
      className="dropdown"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <SimpleButton
        action={[Function]}
        label="DropDown"
        value=""
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "borderColor": "green",
              "color": "green",
            }
          }
        >
          DropDown
        </button>
      </SimpleButton>
    </div>
  </ToggleState>
</ToggleSimpleButton>
`;
